### DISCUSSION TITLE
How can I append a DataTable filter overlay to any HTML element, excluding the default Protal value "body"?

### DISCUSSION THREAD

**Discussion by andrew-cmdltt** [primefaces/community#2318] [PrimeVue]
I know that PrimeVue has a Portal component that is used for appending wrapped elements to any HTML element. For this, we should provide the appendTo property with a string value to find an HTML element by querySelector (#needSelector, .needSelector and so on). I use this property forCalendar, but is there a similar prop for DataTable column filters? It will be vety helpfull for Module Federation where the styles of shared application incapsulated by selector and overlays can't be appended outside the application wrapper element.

**Comment by colinj** [comment]
I'm bumping this question up. I am using the Datatable component in a modal <dialog> element. The filter menu is not visible because the <Portal> component used for the filter menu has the `attachTo` defaulted to `body` and has no way of setting this to the dialog or `self`. This makes filtering of datatables almost unusable if we cannot change the filter match type. The proposal by @andrew-cmdltt and his PR #6071 would solve this problem

**Comment by Loreeel** [comment]
I also think this would be relevant, since it's impossible to define common styles for this filter panel for a specific layout.

Actually, is it possible that this feature will be added? I see threads on Stack Overflow that have been unanswered for over two years.
